url4c41254592_topskychen_AuthTopk_tgz-pJ8-quickhull3d_QuickHull3DTestJ8/src/utility/security/RSA.java

```diff
@@ -0,0 +1,244 @@
+			//BigInteger m = (p.subtract(BigInteger.ONE))
+	        //         .multiply(q.subtract(BigInteger.ONE)).multiply(p).multiply(q);
+			e =	RSA.PRIME_P;
+		    d = new BigInteger("1134521025461381199780628738339629122047482384507794556518214256222766407939240641355301462595725727630403055841439610538167370181176024435785171895624139924670057634156010264713227319017584709542026521251989552610333061298960431812931715338334915441996250974446628953096609645461395560558237259379836497769063763957643894725229107010847869771811148741932466859705490123089946951666542994825250397188744462581617218567386450896987696984867870226835922454146094716321666838622444420976707886718463203408506658584309274617841897944797505918670232362861870371742229872090674318689314242776254164281194707978579029940223");
+			//while(m.gcd(e).intValue() > 1) e = e.add(new BigInteger("2"));
+		    //d = e.modInverse(m);
+		    //System.out.println(e);
+		    //System.out.println(d);
+			DataInputStream dis = null;
+			if(DEBUG){
+				try {
+					dis = new DataInputStream(new FileInputStream("RSA.seeds"));
+				} catch (FileNotFoundException e2) {
+					// TODO Auto-generated catch block
+					e2.printStackTrace();
+				}
+				//dis = new DataInputStream(Thread.currentThread().getContextClassLoader().getResourceAsStream("/RSA.seeds"));
+				int len;
+				try {
+					len = dis.readInt();
+					byte[] data = new byte[len];
+					dis.read(data, 0, len);
+					n = new BigInteger(data);
+					System.out.println("\"" + n + "\"");
+					len = dis.readInt();
+					data = new byte[len];
+					dis.read(data, 0, len);
+					e = new BigInteger(data);
+					System.out.println("\"" + e + "\"");
+					len = dis.readInt();
+					data = new byte[len];
+					dis.read(data, 0, len);
+					d = new BigInteger(data);
+					System.out.println("\"" + d + "\"");
+				} catch (IOException e1) {
+					// TODO Auto-generated catch block
+					e1.printStackTrace();
+				}
+			}
+			return;
+		}
+		SecureRandom r = new SecureRandom();
+	    BigInteger p = new BigInteger(keySize / 2, 100, r);
+	    BigInteger q = new BigInteger(keySize / 2, 100, r);
+	    n = p.multiply(q);
+	    BigInteger m = (p.subtract(BigInteger.ONE))
+	                   .multiply(q.subtract(BigInteger.ONE));
+	    e =	RSA.PRIME_P;
+	    while(m.gcd(e).intValue() > 1) e = e.add(new BigInteger("2"));
+	    d = e.modInverse(m);
+	    try {
+			DataOutputStream dos = new DataOutputStream(new FileOutputStream("RSA.seeds"));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *dos*. Execution paths that do not contain closure statements were detected. Ensure *dos* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).